### PR TITLE
COMMERCE-7236, COMMERCE-8330, COMMERCE-8347

### PR DESCRIPTION
**Pull request by brianikim**
Resending from: https://github.com/liferay-commerce/liferay-portal/pull/1766

https://issues.liferay.com/browse/COMMERCE-7236

@andrea-ale-sbarra 

https://github.com/liferay-commerce/liferay-portal/pull/2484/commits/511d5dddcacb95c0e9083b26840c809bd8aed9b4
`cpInstance.getCPDefinitionId()` has to be used because we are passing in 0 as a parameter in https://github.com/liferay-commerce/liferay-portal/pull/1766/commits/1c8af69e1763b6840e16343b4e7a0d62948e67de

The deprecation was done so we can utilize the method by actually passing in cpDefinitionId in the future for other use-cases.

Also including https://issues.liferay.com/browse/COMMERCE-8330 and https://issues.liferay.com/browse/COMMERCE-8347

due to baseline changes.

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-item-selector-web/src/main/java/com/liferay/commerce/product/item/selector/web/internal/display/context/CommerceChannelItemSelectorViewDisplayContext.java [hunk 1]
```diff
@@ -91,10 +91,10 @@ public PortletURL getPortletURL() {
 		_searchContainer.setOrderByCol(getOrderByCol());
 		_searchContainer.setOrderByType(getOrderByType());
 		_searchContainer.setResultsAndTotal(
-			() -> _commerceChannelService.search(
+			() -> _commerceChannelService.getCommerceChannels(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-item-selector-web/src/main/java/com/liferay/commerce/product/item/selector/web/internal/display/context/CommerceChannelItemSelectorViewDisplayContext.java:94]: why this is no longer named search?

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceLocalServiceImpl.java [hunk 2]
```diff
@@ -574,17 +584,18 @@ public void checkCPInstancesByDisplayDate(long cpDefinitionId)
 			serviceContext.setWorkflowAction(WorkflowConstants.ACTION_PUBLISH);
 
 			CPDefinition cpDefinition =
-				cpDefinitionLocalService.getCPDefinition(cpDefinitionId);
+				cpDefinitionLocalService.getCPDefinition(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceLocalServiceImpl.java:587]: I don't get this, can you explain me why we need this?

**Review by andrea-ale-sbarra — COMMENTED** [review]
@brianikim I left some comments, can you have a look and let me know? Thanks!

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceLocalServiceImpl.java (hunk 2 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceLocalServiceImpl.java:587]: Now that I reviewed with @alessiorendina I got it. I think this change is right, thanks!

File: modules/apps/commerce/commerce-product-item-selector-web/src/main/java/com/liferay/commerce/product/item/selector/web/internal/display/context/CommerceChannelItemSelectorViewDisplayContext.java (hunk 1 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-item-selector-web/src/main/java/com/liferay/commerce/product/item/selector/web/internal/display/context/CommerceChannelItemSelectorViewDisplayContext.java:94]: got it, fine then. Thanks for letting me know!

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Pull request closed without merging**